### PR TITLE
Fix / Update for ShowColor exiting and Color Zoom

### PR DESCRIPTION
**Pull request by fdarosa2663**
These two are semi-related so submitted together.  This update is to handle clicks and holds while we have ShowColor styles active and also to make Color Zoom functionality for all modes use the same controls to reduce confusion.

### CODE REVIEW

File: props/saber_fett263_buttons.h [hunk 1]
```diff
@@ -2024,24 +2033,88 @@ SaberFett263Buttons() : PropBase() {}
   }
 #endif
 
-  void DoColorZoom() {
-    if (color_mode_ == COLOR_LIST) {
+  bool CancelShowColor() {
+    switch (menu_type_) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:2037]: I seem to remember you having a similar list of cases elsewhere in the code.
Would it be possible to break it out into a helper function. Something like "bool isInMenu()" maybe?

File: props/saber_fett263_buttons.h [hunk 2]
```diff
@@ -2024,24 +2033,88 @@ SaberFett263Buttons() : PropBase() {}
   }
 #endif
 
-  void DoColorZoom() {
-    if (color_mode_ == COLOR_LIST) {
+  bool CancelShowColor() {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:2036]: Needs a comment to explain what this does and what it returns.

File: props/saber_fett263_buttons.h [hunk 3]
```diff
@@ -2024,24 +2033,88 @@ SaberFett263Buttons() : PropBase() {}
...
-  bool EndCCColorList() {
-    if (color_mode_ != CC_COLOR_LIST) return false;
-    hybrid_font.PlayCommon(&SFX_ccend);
-    NewColor(1, BASE_COLOR_ARG);
-    current_preset_.Save();
-    show_color_all_.Stop();
-    UpdateStyle();
-    wav_player.Free();
-    return true;
+  bool EndColorZoom() {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:2095]: Needs comment.

File: props/saber_fett263_buttons.h [hunk 4]
```diff
@@ -4563,9 +4613,40 @@ SaberFett263Buttons() : PropBase() {}
...
+      case EVENTID(BUTTON_POWER, EVENT_RELEASED, MODE_ON):
+#ifndef DISABLE_COLOR_CHANGE
+        if (EndColorZoom()) return true;
+        if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) {
+          ToggleColorChangeMode();
+#ifdef FETT263_EDIT_MODE_MENU
+          if (menu_type_ == MENU_COLOR) {
+            menu_type_ = MENU_TOP;
+            MenuSave();
+	    return true;
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:4637]: indentation looks wrong (on github at least)

File: props/saber_fett263_buttons.h (hunk 1 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:2037]: I think you're thinking of the MenuChoice(), MenuIncrement() and MenuUndo() functions although those are specific to actions taken in the menu not sure if a helper function would make it easier or harder to use.

File: props/saber_fett263_buttons.h (hunk 2 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:2037]: added

File: props/saber_fett263_buttons.h (hunk 3 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:2097]: added

File: props/saber_fett263_buttons.h (hunk 4 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h]: fixed (I think) it looks ok on my end

**Pull request merged**